### PR TITLE
Adds xDnsServerPrimaryZone resource

### LINKED ISSUE #5: xDnsServerPrimaryZone Resource
We could also do with a DNS Primary Zone resource. The xDnsARecord resource should also work as-is with primary zones..

### PR DESCRIPTION
**Pull request by iainbrighton**
* Adds new MSFT_xDnsServerPrimaryZone resource
* Fixes #5 
* Adds test coverage
* Updated readme.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,6 +97,30 @@ configuration Sample_xDnsServerZoneTransfer_TransferToAnyServer
...
+        [Parameter()] [ValidateSet('None','NonsecureAndSecure')]
+        [String]$DynamicUpdate = 'None' 
+    )
+    
+    Import-DscResource -module xDnsServer
+    xDnsServerPrimaryZone addPrimaryZone
+    {
+        Ensure        = 'Present'                
+        Name          = $ZoneName
+        DynamicUpdate = $DynamicUpdate
```
Comment (HemantMahawar) [inline README.md]: ZoneFile property is missing in this sample. It is marked as mandatory property in the schema :)

File: DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.psm1 [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        [ValidateSet('None','NonsecureAndSecure')]
+        [System.String]$DynamicUpdate = 'None',
+        
+        [ValidateSet('Present','Absent')]
+        [System.String]$Ensure = 'Present'
+    )
+
+    Assert-Module -ModuleName 'DNSServer';
+    Write-Verbose ($LocalizedData.CheckingZoneMessage -f $Name, $Ensure);
+    $dnsServerZone = Get-DnsServerZone -Name $Name -ErrorAction SilentlyContinue;
```
Comment (HemantMahawar) [inline DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.psm1]: I would prefer that this code goes in Get-TargetResource function which should return the current state of PrimaryZone

File: DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.schema.mof [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[ClassVersion("1.0.0.0"), FriendlyName("xDnsServerPrimaryZone")]
+class MSFT_xDnsServerPrimaryZone : OMI_BaseResource
+{
+    [Key, Description("DNS Server primary zone name")] String Name;
+    [Required, Description("DNS Server primary zone file")] String ZoneFile;
+    [Write, Description("Dynamic zone update option"), ValueMap{"None","NonsecureAndSecure"}, Values{"None","NonsecureAndSecure"}] String DynamicUpdate;
```
Comment (HemantMahawar) [inline DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.schema.mof]: This is missing **Secure** value in the valueMap. Is that by design (will implement later) or an oversight?

File: DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.psm1 [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        [System.String]$DynamicUpdate = 'None',
+        
+        [ValidateSet('Present','Absent')]
+        [System.String]$Ensure = 'Present'
+    )
+
+    $targetResource = @{
+        Name = $Name;
+        ZoneFile = $ZoneFile;
+        DynamicUpdate = $DynamicUpdate;
```
Comment (HemantMahawar) [inline DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.psm1]: This is not finding the current state of Zone's dynamicupdate, but merely coping what user provided. Doesn't sound right?

File: DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.psm1 [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+        [ValidateSet('None','NonsecureAndSecure')]
+        [System.String]$DynamicUpdate = 'None',
+        
+        [ValidateSet('Present','Absent')]
+        [System.String]$Ensure = 'Present'
+    )
+
+    $targetResource = @{
+        Name = $Name;
+        ZoneFile = $ZoneFile;
```
Comment (HemantMahawar) [inline DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.psm1]: This is not reporting the zonefile for the given zone, but reports what user had submitted. Doesn't sound correct ...

File: DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.psm1 (hunk 2 above)
Comment (iainbrighton) [inline DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.psm1]: OK - Will move it!

File: DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.psm1 (hunk 5 above)
Comment (iainbrighton) [inline DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.psm1]: I didn't think you could change this after the zone had been created. However, I've just manually tested it and it appears you can! Will update.

File: DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.psm1 (hunk 4 above)
Comment (iainbrighton) [inline DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.psm1]: I'll add this too.

File: DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.schema.mof (hunk 3 above)
Comment (iainbrighton) [inline DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.schema.mof]: The __Secure__ option/value is only a valid for AD integrated zones.

File: README.md (hunk 1 above)
Comment (iainbrighton) [inline README.md]: Good catch :)

File: README.md [hunk 6]
```diff
@@ -10,12 +10,22 @@ Please check out common DSC Resources [contributing guidelines](https://github.c
...
 * **xDnsServerSecondaryZone** sets a Secondary zone on a given DNS server.
 Secondary zones allow client machine in primary DNS zones to do DNS resolution of machines in the secondary DNS zone.
 * **xDnsServerZoneTransfer** This resource allows a DNS Server zone data to be replicated to another DNS server.
 * **xDnsARecord** This resource allwos for the creation of IPv4 host (A) records against a specific zone on the DNS server
 
 
+### xDnsServerPrimaryZone
+
+* **Name**: Name of the primary DNS zone
+* **ZoneFile**: Name of the primary DNS zone file
```
Comment (TravisEz13) [inline README.md]: might mention that it's optional

File: README.md [hunk 7]
```diff
@@ -87,6 +99,33 @@ configuration Sample_xDnsServerZoneTransfer_TransferToAnyServer
...
+
+```powershell
+configuration Sample_xDnsServerPrimaryZone
+{
+    param
+    (
+        [Parameter(Mandatory)]
+        [String]$ZoneName,
+        [Parameter()] [ValidateNotNullOrEmpty()]
+        [String]$ZoneFile = "$ZoneName.dns",
```
Comment (TravisEz13) [inline README.md]: Would this defaulting be good to move into the resource?  Or is it not standard enough?

File: DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.psm1 (hunk 2 above)
Comment (iainbrighton) [inline DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.psm1]: Done!

File: README.md (hunk 7 above)
Comment (iainbrighton) [inline README.md]: @TravisEz13 When you create a zone, the GUI does default to `ZoneName.dns`. I will make this property optional?

Comment (TravisEz13) [inline README.md]: I think we should either open an issue or fix it now.  changing a parameter from required to optional is not breaking so it doesn't have to be done in this PR.

Comment (iainbrighton) [inline README.md]: I'll fix it now ;)

**Pull request merged**